% Introduction

It took more than 20 years for scientist to understand the  physiological role of MHC molecules in peptide presentation to T lymphocytes. 
MHC molecules were initially defined as antigens that stimulate an organismâ€™s immunologic response to transplanted organs and tissues. MHC proteins are found in all higher vertebrates. 
In humans they are called Human Leukocyte Antigens (HLA).

There are two main classes of MHC molecules. Class I and Class II. The genes which encode the various polypeptides forming the two classes of MHC are placed in the genome in the same chromosome.
Two different clusters encode the two different MHC classes. This means that all the genes encoding the class I MHC molecules (HLA-A, HLA-B and HLA-C) are neighbours in a region of chromosome 6,
while the genes encoding the class II MHC molecules (HLA-DP, HLA-DQ and HLA-DR) are close to each other in another region of the same chromosome.

One of the features of MHC is the exceptional allelic diversity. HLA-A and HLA-B have roughly 1000 and 1600 alleles respectively. 
The evolutionary reason for this diversity would be the benefit in the defence of pathogens. 
There is less than 1\% chance that a MHC molecule presents a peptide and different hosts sample different peptides from same pathogen. 
The more diverse MHC molecule the more diversity in presented peptides. As few humans share the same set of HLA alleles, different persons react differently to a pathogen infection. 
This increases the likelihood that at least some individuals of a population will survive an epidemic.

Class-I molecules are present in every cell but Class-II are found on certain immune cells, like macrophages and B-cells. 
These so called antigen-presenting cells (APCs) ingest microbes, destroy them, and digest them into fragments, which are in turn presented by the MHC molecules on the surface of the cell.

The MHC proteins serve to alert the immune system if a foreign material is present inside a cell.
They achieve this by displaying fragmented pieces of antigens on the host cell surface. These antigens can be self or non-self. 
If a host cell was infected by a bacterium or virus, or was cancerous, it may  display the antigens on its surface.
Cells constantly process endogenous proteins and present them within the context of MHC Class-I. 
Immune effector cells are trained not to react to self peptides during a complex process taking place in the thymus.

Only certain types of peptides bind to certain types of MHC molecules. Most peptides selected by class-I molecules are 8 to 10 residues long
and somewhat longer for class-II molecules. The antigenic peptide is located in a cleft existing between the $\alpha_1$ and $\alpha_2$ regions in the heavy chain of the MHC protein.

A CTL based vaccine must include epitopes specific to each HLA allele in a population, also it must deal with pathogen diversity. As result a vaccine must consist of around 1000 HLA class I epitopes.
Less than 70 HLA alleles have been characterized by binding data. Apart from predicting peptide binders in a protein sequence, it is still  mostly unsolved task of how to select the immunogenic proteins in large pathogens.

The classification of MHC peptide binding is a complex task as the mechanics of binding are not fully understood. 
This makes the problem an ideal task for machine learning methods. We compare binding data classification results from PSSM and the machine learning methods support vector machines (SVM) and artificial neural networks (ANN).

Using amino acid properties such as hydropathy index, side-chain polarity or charge and isoelectric point are not sufficient for classification although they can give clues on which locations on the peptide are more important to successful binding than others. 
Some of these properties are taken into account in the z-score amino acid encoding scheme. Structural properties of the MHC molecule are also important in governing the binding affinity.

Some efforts are being made using molecular dynamics simulation and crystal structure data for MHC peptide binding predictions, although these have not proved successful compared to the sequence based methods.

All of the software tools used in this investigation are part of the course and available at CBS apart from the Weka software we used for the SVM calculations. [reference: http://www.cs.waikato.ac.nz/~ml/weka/]
