\documentclass[presentation]{beamer}   % to compile the presentation
%\documentclass[handout]{beamer}        % to compile 2x2 handouts
\usepackage[ansinew]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{lmodern,textcomp}
\usepackage{breakurl}
\usepackage{graphicx}

\def\supertiny{ \font\supertinyfont = cmr10 at 4pt \relax \supertinyfont} 
\usetheme{Algo}
%\usetheme{Warsaw}

\usepackage[formats]{listings}
\lstdefineformat{C}{% 
	\{=\newline\string\newline\indent,% 
	\}=[;]\newline\noindent\string\newline,% 
	\};=\newline\noindent\string\newline,% 
	;=[\ ]\string\space}
\lstset{language=C}

%\usepackage{/usr/lib64/R/share/texmf/Sweave}
%\usepackage{/Library/Frameworks/R.framework/Resources/share/texmf/Sweave}
\setbeamercovered{transparent}
\AtBeginSection[]
        {
                \begin{frame}<beamer>
                        \frametitle{}
                        \tableofcontents[currentsection]
                \end{frame}
        }

\begin{document}
\pgfdeclareimage[height=1cm]{Biglogo}{dtu_logo}
%\pgfdeclareimage[width=12mm]{footlogo}{partek_logo_s}

\author{Sigmar Stef\`{a}nson and Francesco Favero}
\title[PSSM,ANN,SVM]{Comparison of MHC peptide binding data classification using PSSM, SVM and ANN}
\date{15 Dic. 2010}
\titlegraphic{\pgfuseimage{Biglogo}} % Graphics for title slide
%\logo{\pgfuseimage{footlogo}} % The left logo



\begin{frame}
  \maketitle
\end{frame}

\begin{frame}
  \frametitle{Outline}
  \tableofcontents[currentsection]
\end{frame}
\section{Introduction}
\begin{frame}
  \frametitle{Methods}
  \begin{itemize}
    \item<1> We looked at three methods from the class used for MHC peptide binding prediction.
    \item<2>  The Position specific scoring matrix (PSSM), Support vector machines (SVM) and Artificial neural networks (ANN).
    \item<3> Generating PSSM weight matrix just uses data from positive binders (> 0.426 binding coefficient)
    \item<4> The other two, SVM and ANN are machine learning methods, non-binders also useful.
    \item<5> Pearsons correlation coefficient to evaluate the predictive performance of the methods.
    \item<6> its invariant in terms of location and scale and should therefore be ideal comparing different methods.
	\begin{equation}
		pcc = \frac{ \sum_n{(x-x_m)(y-y_m) } }{ \sqrt{ \sum_n{(x-x_m)^2}\cdot\sum_{n}{(y-y_m)^2} } }
	\end{equation}
  \end{itemize}
\end{frame}

\subsection{Data}
\begin{frame}
  \begin{itemize}
    \item<1> All 35 MHC datasets from course used in PSSM and ANN.
    \item<2>  Looked specifically into datasets containing relatively few binders (B4001) to stress the difference between methods using and not using non-binding data.
    \item<3> Also compared results of smaller datasets.
  \end{itemize}
\end{frame}

\subsection{Methods}
\begin{frame}
 \frametitle{PSSM (PWM)}
 \begin{columns}
  \begin{column}{5cm}
   \begin{block}\centering
    PSSM is used to represent a motif pattern.
    Is a good method to estimate the relevance 
    of the position of an aminoacids in a MHC binding. 
   \end{block}
   \pause
   \begin{block}\centering
   Uses \textit{Pseudo Counts} when few data are available:
    \begin{equation}
     p_a = \frac{\alpha \cdot f_a + \beta \cdot g_a}{\alpha + \beta}
    \end{equation}
   \end{block}
   \pause
  \end{column}
  \begin{column}{5cm}
   \begin{block}\centering
    \textit{Sequence Weighting} can be used to reduce redoundancy:
    \begin{equation}
     w_{k} = \sum_{p}{\frac{1}{r_p \cdot s_p}}
    \end{equation}

   \end{block}
   \pause
   \begin{block}\centering
    Where the information content is:
     \begin{equation}
     I = \log{ 20 } + \sum_{a}{ p_{a}\log{p_{a}} }
    \end{equation}
   \end{block}
  
  
  \end{column}
 \end{columns}

\end{frame}

\begin{frame}
 \frametitle{SVM}
 
 \begin{columns}
  \begin{column}{5cm}
   \begin{block}\centering
 
   \end{block}
   \pause
   \begin{block}\centering

   \end{block}
   \pause
  \end{column}
  \begin{column}{5cm}
   \begin{block}\centering

   \end{block}
   \pause
   \begin{block}\centering
   
   \end{block}
  \end{column}
 \end{columns}
\end{frame}

\begin{frame}
 \frametitle{ANN}
 
  \begin{columns}
  \begin{column}{5cm}
   \begin{block}\centering
    Artificial Neural Networks are non-linear statistical data modeling
   \end{block}
   \pause
   \begin{block}\centering
    ANN is an ideal method to consider the global effects of the 
    peptides in the sequence, not just those in the binding site. 
   \end{block}
   \pause
  \end{column}
  \begin{column}{5cm}
   \begin{block}\centering
    Information Contents similar to PSSM:
    \begin{equation}
     I = \sum_{a,b}{ p_{ab}\log{ \frac{ p_{ab} }{ p_a \cdot p_b } } }
    \end{equation}
   \end{block}
   \pause
   \begin{block}\centering
   For every edge of the neural network layer, a weight $w_i$ is associated:
    \begin{equation}
     o = \sum{x_i \cdot w_i}
    \end{equation}
   \end{block}
  \end{column}
 \end{columns}
\end{frame}


\section{PSSM}
\begin{frame}
 \begin{itemize}
\item<1> Using sequence weighting providest better results in most cases.
\item<2> The B4001 clearly visible, large dataset, low number of binders, PCC around 0.3
\item<3> Small datasets provide PSSM with good prediction performance, pseudo counts helps.
\item<4> Using same datasets in all methods, the PSSM results were (average PCC) 0.30 with and 0.26 without sequence weighting for B4001
\item<5> 0.61 and 0.60 for A3001, 0.75 and 0.77 for A0201.
\end{itemize}
\end{frame}

\begin{frame}
\frametitle{PSSM results for all the 35 Alleles}
 \begin{figure}[ht]
  \begin{center}
   \includegraphics[width=10cm]{fig/pssmLN1.pdf}
  \end{center}
 \end{figure}
\end{frame}

\begin{frame}
\frametitle{PSSM results for all the 35 Alleles}
 \begin{figure}[ht]
  \begin{center}
   \includegraphics[width=10cm]{fig/pssm_annot1.pdf}
  \end{center}
 \end{figure}
\end{frame}
\begin{frame}
\frametitle{PSSM results for all the 35 Alleles}
 \begin{figure}[ht]
  \begin{center}
   \includegraphics[width=10cm]{fig/pssm_annot2.pdf}
  \end{center}
 \end{figure}
\end{frame}
\begin{frame}
\frametitle{PSSM results for all the 35 Alleles}
 \begin{figure}[ht]
  \begin{center}
   \includegraphics[width=10cm]{fig/pssm_annot3.pdf}
  \end{center}
 \end{figure}
\end{frame}

\begin{frame} 
\begin{itemize}
\item<1> Using sequence weighting provides better results in most cases.
\item<2> The B4001 clearly visible, large dataset, low number of binders, PCC just over 0.3
\item<3> Small datasets provide PSSM with good prediction performance, pseudo counts helps.
\end{itemize}
\end{frame}

\begin{frame}

\begin{table}\scriptsize

\begin{center}


\begin{tabular}{rllrrr}
  \hline
 & Param & Allele & Sample & Size & PCC \\ 
  \hline
 & PSSM & A0201 &   0 & 618 & 0.74 \\ 
 & PSSM & A0201 &   1 & 618 & 0.76 \\ 
 & PSSM & A0201 &   2 & 618 & 0.76 \\ 
 & PSSM & A0201 &   3 & 618 & 0.76 \\ 
 & PSSM & A0201 &   4 & 617 & 0.75 \\ 
\hline
 & PSSM SW & A0201 &   0 & 618 & 0.75 \\ 
 & PSSM SW & A0201 &   1 & 618 & 0.78 \\ 
 & PSSM SW & A0201 &   2 & 618 & 0.78 \\ 
 & PSSM SW & A0201 &   3 & 618 & 0.79 \\ 
 & PSSM SW & A0201 &   4 & 617 & 0.76 \\ 
   \hline
\end{tabular}

\end{center}
\end{table}
\end{frame}

\begin{frame}
\begin{table}\scriptsize

\begin{center}


\begin{tabular}{rllrrr}
  \hline
 & Param & Allele & Sample & Size & PCC \\ 
  \hline
   & PSSM & A3001 &   0 & 134 & 0.68 \\ 
   & PSSM & A3001 &   1 & 134 & 0.55 \\ 
   & PSSM & A3001 &   2 & 134 & 0.64 \\ 
   & PSSM & A3001 &   3 & 134 & 0.55 \\ 
   & PSSM & A3001 &   4 & 133 & 0.56 \\ 
  \hline
   & PSSM SW & A3001 &   0 & 134 & 0.70 \\ 
   & PSSM SW & A3001 &   1 & 134 & 0.56 \\ 
   & PSSM SW & A3001 &   2 & 134 & 0.65 \\ 
   & PSSM SW & A3001 &   3 & 134 & 0.56 \\ 
   & PSSM SW & A3001 &   4 & 133 & 0.57 \\ 
   \hline
\end{tabular}

\end{center}
\end{table}
\end{frame}

\begin{frame}
\begin{table}\scriptsize

\begin{center}
\begin{tabular}{rllrrr}
  \hline
 & Param & Allele & Sample & Size & PCC \\ 
  \hline
   & PSSM & B4001 &   0 & 216 & 0.19 \\ 
   & PSSM & B4001 &   1 & 216 & 0.24 \\ 
   & PSSM & B4001 &   2 & 216 & 0.19 \\ 
   & PSSM & B4001 &   3 & 215 & 0.38 \\ 
   & PSSM & B4001 &   4 & 215 & 0.29 \\ 
\hline
   & PSSM SW & B4001 &   0 & 216 & 0.23 \\ 
   & PSSM SW & B4001 &   1 & 216 & 0.29 \\ 
   & PSSM SW & B4001 &   2 & 216 & 0.24 \\ 
   & PSSM SW & B4001 &   3 & 215 & 0.41 \\ 
   & PSSM SW & B4001 &   4 & 215 & 0.32 \\ 
   \hline
\end{tabular}
\end{center}
\end{table}
\end{frame}


\section{SVM}
 \frametitle{SVM}
\begin{frame}
\begin{table}[ht]\scriptsize
\begin{center}
\begin{tabular}{rllrrr}
  \hline
 & Param & Allele & Size & PCC & MAE \\ 
  \hline
 & SVM Pol. 1$^{st}$ d + Sparse & A0201 &   618 & 0.78 & 0.1524 \\ 
 & SVM Pol. 2$^{st}$ degree & A0201 &   618 & 0.756 & 0.1535 \\ 
 & SVM Pol. 1$^{st}$ d + Blosum & A0201 &   618 & 0.7789 & 0.1533 \\ 
 & SVM Pol. 2$^{nd}$ degree & A0201 &   618 & 0.6692 & 0.2047 \\ 
 & SVM Pol. 1$^{st}$ d + z-score & A0201 &   618 & 0.6888 & 0.1802 \\ 
 & SVM Sparse & A3001 &   134 & 0.7412 & 0.1008 \\ 
 & SVM Pol. 1$^{nd}$ d + Blosum & A3001 &   134 & 0.7671 & 0.0945 \\ 
 & SVM Pol. 1$^{nd}$ d + Sparse & B4001 &   216 & 0.4876 & 0.0373 \\ 
 & SVM Pol. 1$^{nd}$ d + Blosum & B4001 &   216 & 0.4456 & 0.0387 \\ 
 & SVM Pol. 1$^{nd}$ d + Zscore & B4001 &   216 & 0.2397 & 0.0400 \\ 
   \hline
\end{tabular}
\end{center}
\end{table}

\end{frame}

\section{ANN}

\begin{frame}
\frametitle{ANN results for all the 35 Alleles}
\begin{figure}[ht]
\begin{center}
\includegraphics[width=10cm]{fig/annBX1.pdf}
\end{center}
\end{figure}
\end{frame}


\begin{frame}
\begin{table}[hb]\scriptsize
\begin{center}

\begin{tabular}{rllrrr}
  \hline
 & Param & Allele & Sample & Size & PCC \\ 
  \hline
   & NN 10HL & A0201 &   0 & 618 & 0.84 \\ 
   & NN 10HL & A0201 &   1 & 618 & 0.87 \\ 
   & NN 10HL & A0201 &   2 & 618 & 0.86 \\ 
   & NN 10HL & A0201 &   3 & 618 & 0.86 \\ 
   & NN 10HL & A0201 &   4 & 617 & 0.87 \\ 
   \hline
   & NN 10HL Blosum & A0201 &   0 & 618 & 0.83 \\ 
   & NN 10HL Blosum & A0201 &   1 & 618 & 0.86 \\ 
   & NN 10HL Blosum & A0201 &   2 & 618 & 0.85 \\ 
   & NN 10HL Blosum & A0201 &   3 & 618 & 0.86 \\ 
   & NN 10HL Blosum & A0201 &   4 & 617 & 0.86 \\ 
   \hline
   & NN 2HL & A0201 &   0 & 618 & 0.84 \\ 
   & NN 2HL & A0201 &   1 & 618 & 0.87 \\ 
   & NN 2HL & A0201 &   2 & 618 & 0.86 \\ 
   & NN 2HL & A0201 &   3 & 618 & 0.86 \\ 
   & NN 2HL & A0201 &   4 & 617 & 0.87 \\ 
   \hline
   & NN 2HL Blosum & A0201 &   0 & 618 & 0.85 \\ 
   & NN 2HL Blosum & A0201 &   1 & 618 & 0.87 \\ 
   & NN 2HL Blosum & A0201 &   2 & 618 & 0.85 \\ 
   & NN 2HL Blosum & A0201 &   3 & 618 & 0.87 \\ 
   & NN 2HL Blosum & A0201 &   4 & 617 & 0.87 \\ 
   \hline
\end{tabular}
\end{center}

\end{table}
\end{frame}

\begin{frame}
\begin{table}[hb]\scriptsize
\begin{center}
\begin{tabular}{rllrrr}
  \hline
 & Param & Allele & Sample & Size & PCC \\ 
  \hline
   & NN 10HL & A3001 &   0 & 134 & 0.81 \\ 
   & NN 10HL & A3001 &   1 & 134 & 0.66 \\ 
   & NN 10HL & A3001 &   2 & 134 & 0.80 \\ 
   & NN 10HL & A3001 &   3 & 134 & 0.67 \\ 
   & NN 10HL & A3001 &   4 & 133 & 0.71 \\ 
\hline
   & NN 10HL Blosum & A3001 &   0 & 134 & 0.82 \\ 
   & NN 10HL Blosum & A3001 &   1 & 134 & 0.65 \\ 
   & NN 10HL Blosum & A3001 &   2 & 134 & 0.78 \\ 
   & NN 10HL Blosum & A3001 &   3 & 134 & 0.68 \\ 
   & NN 10HL Blosum & A3001 &   4 & 133 & 0.62 \\ 
\hline
   & NN 2HL & A3001 &   0 & 134 & 0.82 \\ 
   & NN 2HL & A3001 &   1 & 134 & 0.66 \\ 
   & NN 2HL & A3001 &   2 & 134 & 0.81 \\ 
   & NN 2HL & A3001 &   3 & 134 & 0.67 \\ 
   & NN 2HL & A3001 &   4 & 133 & 0.71 \\ 
\hline
   & NN 2HL Blosum & A3001 &   0 & 134 & 0.84 \\ 
   & NN 2HL Blosum & A3001 &   1 & 134 & 0.66 \\ 
   & NN 2HL Blosum & A3001 &   2 & 134 & 0.80 \\ 
   & NN 2HL Blosum & A3001 &   3 & 134 & 0.68 \\ 
   & NN 2HL Blosum & A3001 &   4 & 133 & 0.60 \\ 
   \hline
\end{tabular}
\end{center}

\end{table}
\end{frame}
\begin{frame}
\begin{table}[hb]\scriptsize
\begin{center}
\begin{tabular}{rllrrr}
  \hline
 & Param & Allele & Sample & Size & PCC \\ 
  \hline
   & NN 10HL & B4001 &   0 & 216 & 0.58 \\ 
   & NN 10HL & B4001 &   1 & 216 & 0.65 \\ 
   & NN 10HL & B4001 &   2 & 216 & 0.60 \\ 
   & NN 10HL & B4001 &   3 & 215 & 0.59 \\ 
   & NN 10HL & B4001 &   4 & 215 & 0.65 \\ 
\hline
   & NN 10HL Blosum & B4001 &   0 & 216 & 0.61 \\ 
   & NN 10HL Blosum & B4001 &   1 & 216 & 0.62 \\ 
   & NN 10HL Blosum & B4001 &   2 & 216 & 0.56 \\ 
   & NN 10HL Blosum & B4001 &   3 & 215 & 0.60 \\ 
   & NN 10HL Blosum & B4001 &   4 & 215 & 0.60 \\ 
\hline
   & NN 2HL & B4001 &   0 & 216 & 0.60 \\ 
   & NN 2HL & B4001 &   1 & 216 & 0.66 \\ 
   & NN 2HL & B4001 &   2 & 216 & 0.63 \\ 
   & NN 2HL & B4001 &   3 & 215 & 0.62 \\ 
   & NN 2HL & B4001 &   4 & 215 & 0.67 \\ 
\hline
   & NN 2HL Blosum & B4001 &   0 & 216 & 0.64 \\ 
   & NN 2HL Blosum & B4001 &   1 & 216 & 0.60 \\ 
   & NN 2HL Blosum & B4001 &   2 & 216 & 0.56 \\ 
   & NN 2HL Blosum & B4001 &   3 & 215 & 0.63 \\ 
   & NN 2HL Blosum & B4001 &   4 & 215 & 0.59 \\ 
   \hline
\end{tabular}
\end{center}
\end{table}
\end{frame}


 \frametitle{SVM}
\begin{frame}
\begin{table}[ht]\scriptsize
\begin{center}
\begin{tabular}{rllrrr}
  \hline
 & Param & Allele & Size & PCC & MAE \\ 
  \hline
 & SVM Pol. 1$^{st}$ d + Sparse & A0201 &   618 & 0.78 & 0.1524 \\ 
 & SVM Pol. 2$^{st}$ degree & A0201 &   618 & 0.756 & 0.1535 \\ 
 & SVM Pol. 1$^{st}$ d + Blosum & A0201 &   618 & 0.7789 & 0.1533 \\ 
 & SVM Pol. 2$^{nd}$ degree & A0201 &   618 & 0.6692 & 0.2047 \\ 
 & SVM Pol. 1$^{st}$ d + z-score & A0201 &   618 & 0.6888 & 0.1802 \\ 
 & SVM Sparse & A3001 &   134 & 0.7412 & 0.1008 \\ 
 & SVM Pol. 1$^{nd}$ d + Blosum & A3001 &   134 & 0.7671 & 0.0945 \\ 
 & SVM Pol. 1$^{nd}$ d + Sparse & B4001 &   216 & 0.4876 & 0.0373 \\ 
 & SVM Pol. 1$^{nd}$ d + Blosum & B4001 &   216 & 0.4456 & 0.0387 \\ 
 & SVM Pol. 1$^{nd}$ d + Zscore & B4001 &   216 & 0.2397 & 0.0400 \\ 
   \hline
\end{tabular}
\end{center}
\end{table}
\end{frame}

\section{Conclusion}
\begin{frame}
\begin{itemize}
\item<1> PSSM method almost as good as best result from SVM using largest dataset A0201 (pcc 0.77 vs 0.78)
\item<2> Maybe bad choose of kernel function/parameters, or PSSM method simply accurate.
\item<3> SVM has the winning in dataset with few binders, best pcc 0.49 for B4001 using SVM compared to 0.3 with PSSM.
\item<4> Also for the relatively small dataset A3001, SVM performs better than PSSM (pcc 0.77 vs 0.61)
\item<5> For the few datasets we tested SVM on, using Blosum encoded data did not provide better results. (Further evidence needed to be able to accurately comment on this)
\item<6> First order polynomial kernel function performed better than $2^{nd}$ order in all cases.
\item<7> Z-score encoded data not performing as well as we would hope as it is based on structural/functional info on the amino acids.
\end{itemize}
\end{frame}

\begin{frame}
\begin{itemize}
\item<1> The ANN's have best overall performance of the three methods.
\item<2> For the large dataset A0201, performance for 2 and 10 hidden layers, with and without blosum matrix showed very similar results, pcc around 0.86
\item<3> This is substancially better than the other methods, even though we were more conservative in terms of the overfitting problem when training the ANN's (no cross-validation used in SVM)
\item<4> The best results for the B4001 dataset was pcc 0.64, also much better than with the other methods. Not much deviation in the results for all types of networks (2/10 hidden layers, blosum/no blosum) (all >= 0.6)
\end{itemize}
\end{frame}

\end{document} 
